### PR TITLE
feat: add support for encoding in loadjob bigquery

### PR DESCRIPTION
**Pull request by hemanshv**

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs [hunk 1]
```diff
@@ -91,6 +91,14 @@ public sealed class CreateLoadJobOptions : JobCreationOptions
         /// </summary>
         public string NullMarker { get; set; }
 
+        /// <summary>
+        /// The character encoding of the data. The supported values are UTF-8
+        /// or ISO-8859-1. The default value is UTF-8. BigQuery decodes the data after the
+        /// raw, binary data has been split using the values of the quote and fieldDelimiter
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs:97]:
```suggestion
        /// raw, binary data has been split using the values of the <see cref="Quote"/> and <see cref="FieldDelimiter"/>
```

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadCsvOptions.cs [hunk 2]
```diff
@@ -87,6 +87,14 @@ public sealed class UploadCsvOptions : JobCreationOptions
         /// </summary>
         public string NullMarker { get; set; }
 
+        /// <summary>
+        /// The character encoding of the data. The supported values are UTF-8
+        /// or ISO-8859-1. The default value is UTF-8. BigQuery decodes the data after the
+        /// raw, binary data has been split using the values of the quote and fieldDelimiter
+        /// properties.
+        /// </summary>
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/UploadCsvOptions.cs:95]: Same comments here as for the similar property in CreateLoadJobOptions.

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs [hunk 3]
```diff
@@ -91,6 +91,14 @@ public sealed class CreateLoadJobOptions : JobCreationOptions
         /// </summary>
         public string NullMarker { get; set; }
 
+        /// <summary>
+        /// The character encoding of the data. The default value is UTF-8. BigQuery decodes the data after the
+        /// raw, binary data has been split using the values of the <see cref="Quote"/> and <see cref="FieldDelimiter"/>
+        /// properties.
+        /// Refer https://cloud.google.com/bigquery/docs/reference/rest/v2/Job#JobConfigurationLoad.FIELDS.encoding for more details.
```
Comment (jskeet) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/CreateLoadJobOptions.cs:98]: Please change this to "See" or "Refer to" - just "refer xyz" doesn't sound quite right.

**Comment by jskeet** [comment]
Any information about whether this applies to JSON or not?

**Comment by jskeet** [comment]
> Any information about whether this applies to JSON or not?

Answered internally - no, we don't believe we should add this to UploadJsonOptions.

**Review by jskeet — APPROVED** [review]
Fine after the minor comment nit.

**Pull request merged**